HW_1/myproject/shopapp/views.py

```diff
@@ -0,0 +1,15 @@
+from django.shortcuts import render, get_object_or_404
+from shopapp.models import Order, User, Product
+import datetime
+
+# Create your views here.
+
+def get_order_by_user(request, customer_pk):
+    customer = get_object_or_404(User, pk=customer_pk)
+    week_date = datetime.datetime.today() - datetime.timedelta(days=7)
+    month_date = datetime.datetime.today() - datetime.timedelta(days=30)
+    year_date = datetime.datetime.today() - datetime.timedelta(days=365)
+    orders_week = Order.objects.filter(customer=customer, date_order__gte=week_date).all()
```

Comment:
Можно сделать иначе. Так как у нас таблицы заказов и продуктов связаны между собой, к таблице заказов можно обращаться через таблицу с продуктами. Таким образом, можно извлечь все продукты, которые содержатся в определенных заказах. Вот пример, как это можно было сделать:
```suggestion
    orders_week = Product.objects.filter(order__customer=customer, order__date_order__gte=week_date).distinct()
```